main_test.go

```diff
@@ -1,31 +1,65 @@
 package main
 
 import (
+	"embed"
+	"io/fs"
 	"os"
 	"testing"
 
 	"github.com/rogpeppe/go-internal/testscript"
+	"github.com/stretchr/testify/require"
 
 	"github.com/smartcontractkit/chainlink/v2/core"
 	v2 "github.com/smartcontractkit/chainlink/v2/core/config/v2"
 	"github.com/smartcontractkit/chainlink/v2/core/static"
 )
 
+//go:embed testdata/**
+var testFs embed.FS
+
 func TestMain(m *testing.M) {
 	os.Exit(testscript.RunMain(m, map[string]func() int{
 		"chainlink": core.Main,
 	}))
 }
 
 func TestScripts(t *testing.T) {
-	testscript.Run(t, testscript.Params{
-		Dir: "testdata/scripts",
-		Setup: func(env *testscript.Env) error {
-			env.Setenv(string(v2.EnvDev), "true")
-			env.Setenv("HOME", "$WORK/home")
-			env.Setenv("VERSION", static.Version)
-			env.Setenv("COMMIT_SHA", static.Sha)
-			return nil
-		},
-	})
+	t.Parallel()
+	testDataRootDir := "testdata/scripts"
+
+	visitFn := func(path string, d os.DirEntry, err error) error {
+		t.Logf("path %s", path)
+		if err != nil {
+			return err
+		}
+		if d.IsDir() && hasScripts(t, path) {
```

Comment:
You might get some speedup from returning `fs.SkipDir` when `!hasScripts`.

Reply:
Eh, but then we can't have dirs with only dirs.

Reply:
right, can't skip b/c child may have scripts